avocado/utils/pci.py

```diff
@@ -97,20 +98,25 @@ def get_disks_in_pci_address(pci_address):
     return disk_list
 
 
-def get_nics_in_pci_address(pci_address):
+def get_interface_in_pci_address(pci_address, pci_class):
     """
-    Gets network interface(nic) in a PCI address.
+    Gets interface in a PCI address.
 
     :param pci_address: Any segment of a PCI address (1f, 0000:00:1f, ...)
 
-    :return: list of network interfaces in a PCI address.
+    :param clas: Adapter type (FC(fc_host), FCoE(net), NIC(net), SCSI(scsi)..)
+
+    :return: list of generic interfaces in a PCI address.
     """
-    iface_path = "/sys/class/net/"
-    net_interfaces_list = []
-    for iface in os.listdir(iface_path):
-        if pci_address in os.readlink("%s%s" % (iface_path, iface)):
-            net_interfaces_list.append(iface)
-    return net_interfaces_list
+    pci_class_path = "/sys/class/%s/" % pci_class
+    if not os.path.isdir(pci_class_path):
+        logging.debug("please pass valid class name")
```

Comment:
Actually this is a failure, isn't it? I'd rather see `assert os.path.isdir(pci_class_path), "Incorrect class name, valid ones are %s" % os.path.listdir("/sys/class")` or raising a `ValueError` (the later is better as `ValueError` better describes the failure)